addons/mail/models/discuss/discuss_channel.py

```diff
@@ -705,6 +706,26 @@ def _channel_message_notifications(self, message, message_format=False):
     #   - when a message is posted on a channel (to the channel, using _notify() method)
     # ------------------------------------------------------------
 
+    def set_allow_public_upload(self, allow_public_upload):
+        """ Set the allow_public_upload field of the channel to the given
+        value and notify the channel members.
+
+        :param bool allow_public_upload: Whether to allow public upload.
+        """
+        self.ensure_one()
+        self.allow_public_upload = allow_public_upload
```

Comment:
If the current value is already equal the given value, the method could be made to do nothing.

Reply:
This method edit boolean field, wouldn't it be easier to just make it a toggle and remove the `allow_public_upload` param?
Maybe the UI should be a bit edited for this through?

Reply:
toggle is very prone to race condition, see other comment I posted to change it to a pram